node_monitor/node_provider_db.py

```diff
@@ -315,17 +315,44 @@ def get_channels(self) -> List[Tuple[Any, ...]]:
     ##############################################
     ## CRUD :: TABLE node_label_lookup
     
-    def _insert_node_label(self) -> None:
-        """Inserts a node label into the node_label_lookup table."""
-        raise NotImplementedError
+    def _insert_node_label(self, node_id: str, node_label: str) -> None:            
+        """Inserts or updates a node label into the node_label_lookup table."""
+        query = """
+            INSERT INTO node_label_lookup (node_id, node_label)
+            VALUES (%s, %s)  
+            ON CONFLICT (node_id) DO UPDATE SET
+                node_label = EXCLUDED.node_label
+        """
+        values = (node_label, node_id)
+        self.connect()
+        assert self.conn is not None
+        with self.conn.cursor() as cur:
+            cur.execute(query, values)
+        self.disconnect()
     
-    def _delete_node_label(self) -> None:
+    def _delete_node_label(self, node_label: str) -> None:
```

Comment:
This should work on `node_id` and not `node_label`

Reply:
I saw that you already changed this so I will resolve this comment :) Thanks for spotting and fixing it!!